app/sigverify_test.go

```diff
@@ -0,0 +1,134 @@
+package app
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	kmultisig "github.com/cosmos/cosmos-sdk/crypto/keys/multisig"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256r1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/crypto/types/multisig"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/stretchr/testify/require"
+	"github.com/tharsis/ethermint/crypto/ethsecp256k1"
+)
+
+func generatePubKeysAndSignatures(n int, msg []byte, _ bool) (pubkeys []cryptotypes.PubKey, signatures [][]byte) {
+	pubkeys = make([]cryptotypes.PubKey, n)
+	signatures = make([][]byte, n)
+	for i := 0; i < n; i++ {
+		privkey, _ := ethsecp256k1.GenerateKey()
+		pubkeys[i] = privkey.PubKey()
+		signatures[i], _ = privkey.Sign(msg)
+	}
+	return
+}
+
+func expectedGasCostByKeys(pubkeys []cryptotypes.PubKey) uint64 {
+	cost := uint64(0)
+	for _, pubkey := range pubkeys {
+		pubkeyType := strings.ToLower(fmt.Sprintf("%T", pubkey))
+		switch {
+		case strings.Contains(pubkeyType, "ed25519"):
+			cost += authtypes.DefaultSigVerifyCostED25519
+		case strings.Contains(pubkeyType, "ethsecp256k1"):
+			cost += secp256k1VerifyCost
+		default:
+			panic("unexpected key type")
+		}
+	}
+	return cost
+}
+func TestConsumeSignatureVerificationGas(t *testing.T) {
+	params := authtypes.DefaultParams()
+	msg := []byte{1, 2, 3, 4}
+	cdc := simapp.MakeTestEncodingConfig().Amino
```

Comment:
for the encoding config, let's use the following (see `cmd/evmosd/root.go` for ref) 

```go
encodingConfig := encoding.MakeConfig(app.ModuleBasics)
```